agent/structs/structs.go

```diff
@@ -1153,6 +1153,11 @@ func (s *NodeService) Validate() error {
 				"Proxy.DestinationServiceName must be non-empty for Connect proxy "+
 					"services"))
 		}
+		if strings.TrimSpace(s.Proxy.DestinationServiceName) == WildcardSpecifier {
```

Comment:
For my own understanding, what is the reason for trimming spaces here? I don't think I've seen that done in other places.

Reply:
It was copied from the empty string check above, going to remove it since we're checking against "*".